crates/solvers/src/domain/legacy.rs

```diff
@@ -0,0 +1,68 @@
+//! Legacy HTTP solver adapter implementation.
+//!
+//! In order to faciliate the transition from the legacy HTTP solver API to the
+//! new HTTP API, we provide a solver "wrapper" that just marshals the API
+//! request types to and from the legacy format.
+
+use {
+    crate::{
+        boundary::legacy,
+        domain::{auction, eth, solution, Solver},
+        infra::config::legacy::LegacyConfig,
+    },
+    anyhow::Result,
+    futures::future::{BoxFuture, FutureExt},
+    shared::http_solver::{DefaultHttpSolverApi, HttpSolverApi, SolverConfig},
+};
+
+pub struct Legacy {
+    solver: DefaultHttpSolverApi,
+    weth: eth::WethAddress,
```

Comment:
Sorry for the back-and-forth 🙈.

So `solver: DefaultHttpSolverApi` I would hide in the `boundary` module so that types from the `shared` crate don't leak into the domain.

I really think that the `Legacy` type you had in the `boundary` module was exactly what I would expect to be there - the only change I was suggesting was to change:

```rust
pub use crate::boundary::legacy::Legacy;
```

into:

```rust
pub type Legacy(boundary::legacy::Legacy);

impl Legacy {
    pub fn new(...) -> Self {
        Self(boundary::legacy::Legacy::new(...))
    }
}

impl Solver for Legacy {
    ...
}
```

Reply:
I hope now it's how you meant it. I was not sure about leaving the `solve` method in the boundary directory because the boundary could either be seen as giving all the tools to convert the types back and forth or like it's implementing the old solving strategy.
But by having the "old" solve function in the boundary we have to expose fewer functions/types which seemed like a good thing to me.